File: POSN Gen 16/KVIS Camp/Test/1020 - Palindrome.cpp
#include <bits/stdc++.h>
using namespace std;
char a[210];
int main(){
    int i,len;
    scanf(" %s",a);
    len = strlen(a);
    for(i=0;i<len/2;i++){
        if(isalpha(a[i]))
            a[i] = tolower(a[i]);
        if(isalpha(a[len-i-1]))
            a[len-i-1] = tolower(a[len-i-1]);
        if(a[i]!=a[len-i-1]){
            printf("No\n");
            return 0;
        }
    }
    if(len%2==0) printf("Palindrome\n");
    else printf("Double Palindrome\n");
    return 0;
}

File: Python/1020_Palindrome.py
a = input()
ch = 0
strlen = len(a)
a = a.upper()
for i in range(strlen//2) :
    if a[i] != a[strlen-i-1] : break
    if i == strlen//2-1 : ch = 1
if ch == 1 :
    for i in range(strlen//4) :
        if a[i] != a[strlen//2-i-1] : break
        if i == strlen//4-1 : ch = 2
if ch == 0 : print('No')
elif ch == 1 : print('Palindrome')
elif ch == 2 : print('Double Palindrome')